Gobabygo/Gobabygo.ino

```diff
@@ -37,71 +40,89 @@ float correctForDeadzone(float axis, float deadzone) {
 }
 
 
-// This funtion gradually accelerates a value toward a target
-float accelerateToValue(float currentVal, float targetVal, int timeElapsed, float acceleration) {
-  return currentVal + (targetVal - currentVal) * timeElapsed * acceleration;
+// This function will cap the car's acceleration
+float safeAcceleration(float targetVal, float currentVal, float maxAcceleration) {
+  if (abs(targetVal) - abs(currentVal) < 0) { //faster going toward 0 than away
+    maxAcceleration *= decelerationScaler;
+  }
+  
+  if (abs(targetVal - currentVal) <= maxAcceleration){
+    return targetVal;
+  }
+  else if (targetVal - currentVal > maxAcceleration){
+    return currentVal + maxAcceleration;
+  }
+  else {
+    return currentVal - maxAcceleration;
+  }
 }
 
+/*===================================================================================*/
 
 void setup() {
   // put your setup code here, to run once:
   Serial.begin(9600);
+  pinMode(joyStickXPin, INPUT);
+  pinMode(joyStickYPin, INPUT);
   pinMode(rightMotorPin, OUTPUT);
   pinMode(leftMotorPin, OUTPUT);
+
+  rightMotorController.attach(rightMotorPin);
+  rightMotorController.writeMicroseconds(1500);
+  leftMotorController.attach(leftMotorPin);
+  leftMotorController.writeMicroseconds(1500);
+  
+  delay(3000);
 }
 
 
 void loop() {
   // put your main code here, to run repeatedly:
   currentTime = millis();
   timeSincePrint += currentTime - previousTime;
-  timeSinceAcceleration = currentTime - previousTime;
+  timeSinceMotorUpdate += currentTime - previousTime;
   previousTime = currentTime;
   
-  joyXVal = analogRead(joyStickXPin);
-  joyXVal = 2/1023 * joyXVal - 1;
-  joyXVal = correctForDeadzone(joyXVal, deadzone);
-  joyXVal = accelerateToValue(joyXVal, previousJoyXVal, timeSinceAcceleration, accelerationPerMs);
-  previousJoyXVal = joyXVal;
-  
-  joyYVal = analogRead(joyStickYPin);
-  joyYVal = 2/1023 * joyYVal -1;
-  joyYVal = correctForDeadzone(joyYVal, deadzone);
-  joyYVal = accelerateToValue(joyYVal, previousJoyYVal, timeSinceAcceleration, accelerationPerMs);
-  previousJoyYVal = joyYVal;
-  
-  if (joyYVal < 0) {
-    joyYVal *= backwardPower;
+  if (timeSinceMotorUpdate >= motorLoopTime) {
+    timeSinceMotorUpdate = 0;
+    rawJoyXVal = analogRead(joyStickXPin);
+    targetJoyXVal = 2.0/1023.0 * rawJoyXVal - 1;
+    targetJoyXVal = correctForDeadzone(targetJoyXVal, deadzone);
+    joyXVal = safeAcceleration(targetJoyXVal, joyXVal, motorLoopTime*accelerationPerMs);
+    
+    rawJoyYVal = analogRead(joyStickYPin);
+    targetJoyYVal = 2.0/1023.0 * rawJoyYVal -1;
+    targetJoyYVal = correctForDeadzone(targetJoyYVal, deadzone);
+    if (targetJoyXVal < 0) {
+      targetJoyXVal *= backwardPower;
+    }
```

Comment:
The backward power scaling still is being applied to the X direction instead of the Y direction.